src/toolkit/res_buf.h

```diff
@@ -243,7 +272,7 @@ class ResBuf {
   ///
   /// @throws KeyError the resource object to be pushed is already present
   /// in the buffer.
-  void Push(Resource::Ptr r) {
+  void Push(Resource::Ptr r, bool bulk_storage = true) {
```

Comment:
Given thoughts about a true bulk storage `ResBuf` that squashes everything upon `Push`, should we call this `unpacakged` instead of `bulk_storage`?